internal/pkg/cli/pipeline_init.go

```diff
@@ -434,13 +437,17 @@ func (url ccRepoURL) parse() (ccRepoDetails, error) {
 	case strings.HasPrefix(urlString, "codecommit::"):
 		parsedURL := strings.Split(urlString, ":")
 		region = parsedURL[2]
+	case strings.HasPrefix(urlString, "codecommit://"):
+		// Use default profile region.
 	default:
 		return ccRepoDetails{}, fmt.Errorf("unknown CodeCommit URL format: %s", url)
 	}
-	// Double-check that parsed results is a valid region. Source: https://www.regextester.com/109163
-	match, _ := regexp.MatchString(`(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d`, region)
-	if !match {
-		return ccRepoDetails{}, fmt.Errorf("unable to parse the AWS region from %s", url)
+	if region != "" {
+		// Double-check that parsed results is a valid region. Source: https://www.regextester.com/109163
+		match, _ := regexp.MatchString(`(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d`, region)
```

Comment:
Do we need ability to parse these during `pipeline update` as well? https://github.com/aws/copilot-cli/blob/e67a24e90b02eb501eb2f09506bb7460686aa607/internal/pkg/deploy/pipeline.go#L34

Reply:
Good callout! This seems to be another issue which is more about our manifest design. Posted a new design for tracking this https://github.com/aws/copilot-cli/issues/3021